man/systemd.exec.xml

```diff
@@ -2120,6 +2120,31 @@ BindReadOnlyPaths=/var/lib/systemd</programlisting>
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>MemoryTHPDisable=</varname></term>
+
+        <listitem><para>Transparent Hugepages (THP) is a Linux kernel feature that manages memory
+        using larger pages (2MB on x86, compared to the default 4KB). The main goal is to improve memory management
+        efficiency and system performance, especially for memory-intensive applications.
+        However, it can cause drawbacks in some scenarios, such as memory regression and latency spikes.
+        THP policy is governed for the entire system via <filename>/sys/kernel/mm/transparent_hugepage/enabled</filename>.
+        However, it can be overridden for individual workloads via prctl(2) call.
+        MemoryTHPDisable is used to disable THPs at exec-invoke to stop providing THPs for workloads where the
+        drawbacks outweigh the advantages.
+        When set to 1, MemoryTHPDisable disables THPs completely for the process, irrespecitive of global THP controls.
+        When set to 2, MemoryTHPDisable disables THPs for the process except when specifically madvised by the process
```

Comment:
I don't think that exposing the numerical values accepted by the `prctl()` op is the best user interface. Also, having an option that takes a positive value to disable something is not great. What about a `MemoryTHP=` option that takes a enum with values `disable`, `madvice`, and (optinally) `default`?